### ISSUE TITLE
Some combination of structs and functions leads to compiler crash

### ISSUE THREAD

**Issue by onovytskyi** [microsoft/DirectXShaderCompiler#2526]
Hello,
one more code snippet to crash the compiler
```
struct Resources
{

};

struct ResourceContainer
{
    Resources resources;
};

struct Data
{
    float constants;
    ResourceContainer container;
};

float Calculate(Resources resources)
{
    return 0.0f;
}

static const Data data;
static const float result = Calculate(data.container.resources);

float4 main() : SV_Target
{
    return (float4)0;
}
```

Cheers,
Oleksandr

**Comment by hekota** [comment]
Thanks for the report, we’ll look into it.

**Comment by pow2clk** [comment]
I think this is a dupe of #3770 which was fixed by #3827. Either way, it doesn't repro anymore!
